### PR TITLE
Trigger builds on package upload

### PR DESCRIPTION
**Pull request by chefsalim**
This change adds support for triggering reverse dependency builds when a 'core' package is uploaded to the depot. 

Signed-off-by: Salim Alam <salam@chef.io>

source 1

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -694,6 +702,25 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
...
                    });
 
+        // Schedule re-build of dependent packages (if requested)
+        // Currently, we only do dep builds of 'core' packages
+        if schedule_build && ident.get_origin() == "core" {
+            let mut conn = Broker::connect().unwrap();
+
+            let mut request = GroupCreate::new();
+            request.set_origin(String::from(ident.get_origin().to_string()));
+            request.set_package(String::from(ident.get_name().to_string()));
```
Comment (georgemarshall) [inline components/builder-depot/src/server.rs]: This seems like redundant String creation.

File: components/builder-depot/src/server.rs [hunk 2]
```diff
@@ -1515,6 +1542,13 @@ pub fn routes<M: BeforeMiddleware + Clone>(insecure: bool, basic: M, worker: M)
...
             }
         },
+        package_upload_bldr: post "/pkgs/internal/:origin/:pkg/:version/:release" => {
+            if insecure {
+                XHandler::new(upload_package_bldr)
+            } else {
+                XHandler::new(upload_package_bldr).before(basic.clone())
+            }
+        },
         schedule: post "/pkgs/schedule/:origin/:pkg" => {
```
Comment (mwrock) [inline components/builder-depot/src/server.rs]: should we add a `schedule_build` query parameter to the url and then handle this all in the same handler?

File: components/builder-depot/src/server.rs (hunk 1 above)
Comment (chefsalim) [inline components/builder-depot/src/server.rs]: Yep, good catch - will fix.

File: components/builder-depot/src/server.rs (hunk 2 above)
Comment (chefsalim) [inline components/builder-depot/src/server.rs]: Oh yeah, I like that better... :)

**Comment by reset** [comment]
@thesentinels approve

gif-keyboard-17061753109751822984

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/218703248).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/218703248).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**